Independent telemetry interval

dsabeti opened the pull request:
The previous implementation coupled the telemetry logging interval with the system metrics interval. This commit creates a separate Ifrit process for telemetry logging so that it can be configured independently.

dsabeti committed:
Factor out countIPTablesRules into its own package

dsabeti committed:
Rename poller => pollers

In a future commit, we're going to add a new kind of poller. This commit establishes the 'pollers' package as a collection of poller types. This commit includes a rename of poller.go => system_metrics_poller.go to reflect that.

dsabeti committed:
Add a telemetry poller that can operate on a different poll interval

dsabeti committed:
Make telemetry polling interval configurable

ebroberson commented on src/code.cloudfoundry.org/netmon/config/config.go:
```diff
@@ -37,6 +39,9 @@ func (n Netmon) ParseLogLevel() (lager.LogLevel, error) {
 }
 
 func (c *Netmon) Validate() error {
+	if c.TelemetryEnabled && c.TelemetryInterval == 0 {
```
What happens if c.TelemetryInterval is <0, or is this something that we need to even worry about?

dsabeti committed:
Return an error if telemetry_interval is negative

ebroberson reviewed (approved):
LGTM

geofffranks committed:
Rename networkstatsfetcher to networkstats, with a fetcher inside

Signed-off-by: Geoff Franks <gfranks@vmware.com>
Signed-off-by: David Sabeti <sabetid@vmware.com>

dsabeti committed:
Emit aggregate statistics for telemetry

Specifically, emit maximum, average, and minimumn, as well as the time interval over which those statistics were aggregated

Signed-off-by: Geoff Franks <gfranks@vmware.com>
Signed-off-by: Brandon Roberson <broberson@vmware.com>

ebroberson committed:
Fix context title

Signed-off-by: David Sabeti <sabetid@vmware.com>
